tools/OpenSSLEngine.cpp

```diff
@@ -394,13 +415,17 @@ static EVP_PKEY* tob_engine_load_pubkey(ENGINE* engine, bool signing) {
     free(cert);
   }
 
+  return cert_x509;
```

Comment:
The caller is responsible for cleanup, both in our code here as well as within the caller when used with the LOAD_CERT_CTRL engine command.